pypeit/flatfield.py

```diff
@@ -1317,10 +1317,28 @@ def fit(self, spat_illum_only=False, doqa=True, debug=False):
             # Construct the pixel flat
             #trimmed_slitid_img_anew = self.slits.slit_img(pad=-trim, slitidx=slit_idx)
             #onslit_trimmed_anew = trimmed_slitid_img_anew == slit_spat
-            self.mspixelflat[onslit_tweak] = rawflat[onslit_tweak]/self.flat_model[onslit_tweak]
+            self.mspixelflat[onslit_tweak] = rawflat[onslit_tweak] * utils.inverse(self.flat_model[onslit_tweak])
             # TODO: Add some code here to treat the edges and places where fits
             #  go bad?
 
+            # Check for infinities and NaNs
```

Comment:
This is really nit-picking...but you could add these checks for infinities and NaNs + unrealistically high or low values of the model before filling `self.mspixelflat[onslit_tweak]` in line 1320, so that you will not need to set 
`self.mspixelflat[np.where(onslit_tweak)[0][winfnan]] = 1.0`
and 
`self.mspixelflat[np.where(onslit_tweak)[0][whilo]] = 1.0`

Reply:
This is unrelated, but I would like to ask you something because you understand the flat field code better than me. Here in the `fit()` method, there are several calls to `self.slits.slit_img()` and `self.slits.spatial_coordinate_image()` where the spatial flexure is a parameter, but it's never provided. While, for example, when `fit2illumflat()` is used, the spatial flexure is usually provided. 
What do you think about this? Should the spatial flexure always used?

Reply:
Thanks for your answer @rcooke-ast. It is a bit more clear now and indeed I hadn't thought how the tweaking of the slits affect the spatial flexure.